### PR TITLE
Call out semver expectations for modules

### PR DESCRIPTION
**Pull request by cheddar**
This is the documentation change for https://github.com/druid-io/druid/pull/3632

The author of #3632 is out on vacation and I cannot edit his personal branch, so this is a PR to do it on the side (it can also make sense as a PR)

### CODE REVIEW

**Comment by nishantmonu51** [comment]
:+1:

File: docs/content/development/modules.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Druid uses a module system that allows for the addition of extensions at runtime
 
 ## Writing your own extensions
 
-Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of a few things.
+Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of the things listed below.  This means that we honor semantic versioning for changes that affect the interfaces called out on this page, but other interfaces are deemed "internal" anad can be changed in an incompatible manner even between patch releases.
```
Comment (fjy) [inline docs/content/development/modules.md]: typo: and not anad

Comment (gianm) [inline docs/content/development/modules.md]: "anad" (spelling)

Comment (gianm) [inline docs/content/development/modules.md]: @cheddar can you clarify what "honor semantic versioning" means to us? IIRC, the "official" semver thingy says that no guarantees whatsoever are made if the version number starts with "0". I think we want to say something else, like: so long as the version number starts with 0, these interfaces will remain stable within a 0.X line.

Comment (gianm) [inline docs/content/development/modules.md]: I'm referring to this on http://semver.org/

> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

We're offering something stronger than that.

**Comment by cheddar** [comment]
@gianm on the semver thing.  Generally, we tend to do backwards incompatible changes through an increment of the minor version right now because we are still "0.x".  I consider this us still following a form of semver.  

I intended the "honor semver" comment to basically just reflect to how we version everything else.  If your comment is that we don't have great documentation for how we handle semver, then I think that we should update the version page (http://druid.io/docs/latest/development/versioning.html) instead of hiding it here.  I will go ahead and add a link to that from here though.

**Comment by gianm** [comment]
@cheddar, good point, I forgot about that page. Cross links between the two pages is good enough for me.

**Pull request merged**